libbeat/metric/system/diskio/diskstat_linux.go

```diff
@@ -104,17 +130,17 @@ func (stat *IOStat) CalcIOStatistics(counter disk.IOCountersStat) (IOMetric, err
 	result.WriteRequestCountPerSec = perSec(wrIOs)
 	result.ReadBytesPerSec = perSec(rdBytes)
 	result.WriteBytesPerSec = perSec(wrBytes)
-	result.AvgRequestSize = size
-	result.AvgQueueSize = queue
-	result.AvgAwaitTime = wait
+	result.AvgRequestSize = common.Round(size, common.DefaultDecimalPlacesCount)
+	result.AvgQueueSize = common.Round(queue, common.DefaultDecimalPlacesCount)
+	result.AvgAwaitTime = common.Round(wait, common.DefaultDecimalPlacesCount)
 	if rdIOs > 0 {
-		result.AvgReadAwaitTime = float64(rdTicks) / float64(rdIOs)
+		result.AvgReadAwaitTime = common.Round(float64(rdTicks)/float64(rdIOs), common.DefaultDecimalPlacesCount)
 	}
 	if wrIOs > 0 {
-		result.AvgWriteAwaitTime = float64(wrTicks) / float64(wrIOs)
+		result.AvgWriteAwaitTime = common.Round(float64(wrTicks)/float64(wrIOs), common.DefaultDecimalPlacesCount)
 	}
-	result.AvgServiceTime = svct
-	result.BusyPct = 100.0 * float64(ticks) / deltams
+	result.AvgServiceTime = common.Round(svct, common.DefaultDecimalPlacesCount)
```

Comment:
`svct` is only conditionally not zero per this block above

```go
	svct := float64(0)
	if nIOs > 0 {
		size = float64(nBytes) / float64(nIOs)
		wait = float64(nTicks) / float64(nIOs)
		svct = float64(ticks) / float64(nIOs)
	}
```

Reply:
Not sure what you're asking? `common.Round` should handle zero values just fine?

Reply:
Yes, sorry. I meant to post a comment when I resolved this. I though it was being used in division because every other value I looked at was. I just mis-read the code trying to pattern match for errors.